client/relay-chain-inprocess-interface/src/lib.rs

```diff
@@ -361,11 +360,6 @@ pub fn build_inprocess_relay_chain(
 	task_manager: &mut TaskManager,
 	hwbench: Option<sc_sysinfo::HwBench>,
 ) -> RelayChainResult<(Arc<(dyn RelayChainInterface + 'static)>, Option<CollatorPair>)> {
-	// This is essentially a hack, but we want to ensure that we send the correct node version
-	// to the telemetry.
-	polkadot_config.impl_version = polkadot_cli::Cli::impl_version();
-	polkadot_config.impl_name = polkadot_cli::Cli::impl_name();
```

Comment:
Did you read these comments? This wasn't done without any reason.

Reply:
Need the relay-chain-inprocess-interface to expose its version number. That can be done as a separate polkadot PR.